### ISSUE TITLE
`nix print-dev-env` doesn't handle derivations without `outputs` set

### ISSUE THREAD

**Issue by emilazy** [NixOS/nix#7200]
**Describe the bug**

[This part of `get-env.sh`](https://github.com/NixOS/nix/blob/b3d2a05c59266688aa904d5fb326394cbb7e9e90/src/nix/get-env.sh#L117-L124) assumes `outputs` is set, but it's not required if you manually construct a derivation with the `derivation` primitive rather than using `stdenv.mkDerivation` from nixpkgs.

**Steps To Reproduce**

Run `nix print-dev-env` on a raw `derivation { ... }` without `outputs` set.

**Expected behavior**

If `outputs` is unset, it should default to containing only `out`, like Nix itself does.

**`nix-env --version` output**

```
nix-env (Nix) 2.11.0
```

**Comment by thufschmitt** [comment]
Yes, `nix develop` (and `nix print-dev-env`) unfortunately do several assumptions on the derivation it is called on (at least this and assuming that `runHook` exist).

For this particular one, I think we could just set it to `out` if unset like you say. Adding something like [this](https://github.com/numtide/devshell/blob/c8ce8ed81726079c398f5f29c4b68a7d6a3c2fa2/nix/mkNakedShell.nix#L13) in `get-env.sh` (after `source $stdenv/setup`) should do it.

Wanna give it a shot?
